### ISSUE TITLE
kdeconnect-kde: use stable versions instead of ones from CI

### ISSUE THREAD

**Issue by yan12125** [AndreAugustoAAQ/chocolatey-packages#2]
Hi, thanks a lot for maintaining this package! Here is a minor issue: I often got download failures during upgrades. For example, the current versions gives:

```
You have kdeconnect-kde v22.04.0.952 installed. Version 22.04.1.973 is available based on your source(s).
Progress: Downloading kdeconnect-kde 22.04.1.973... 100%

kdeconnect-kde v22.04.1.973 [Approved]
kdeconnect-kde package files upgrade completed. Performing other installation steps.
The package kdeconnect-kde wants to run 'ChocolateyInstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider:
choco feature enable -n allowGlobalConfirmation
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint): y

Downloading kdeconnect-kde
  from 'https://binary-factory.kde.org/job/kdeconnect-kde_Release_win64/973/artifact/kdeconnect-kde-22.04.1-973-windows-msvc2019_64-cl.exe'
Progress: 46% - Saving 29.41 MB of 63.43 MB
Download of kdeconnect-kde-22.04.1-973-windows-msvc2019_64-cl.exe (63.43 MB) completed.
ERROR: Chocolatey expected a file at 'C:\Users\yen\AppData\Local\Temp\chocolatey\kdeconnect-kde\22.04.1.973\kdeconnect-kde-22.04.1-973-windows-msvc2019_64-cl.exe' to be of length '66510288' but the length was '30843341'.
The upgrade of kdeconnect-kde was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\kdeconnect-kde\tools\ChocolateyInstall.ps1'.
 See log for details.
 Unsuccessful operation for kdeconnect-kde.
  Rollback to previous version (package files only)?([Y]es/[N]o): y
```

I noticed that https://binary-factory.kde.org/ runs Jenkins CI. I guess those files are auto-generated for testing, not for daily-use. Could you consider use "Offline Installers" mentioned https://kdeconnect.kde.org/download.html instead?

**Comment by yan12125** [comment]
Hi @AndreAugustoAAQ, mind to have a look?

**Comment by AndreAugustoDev** [comment]
Hi @yan12125, thanks for the feedback!
Yes, indeed their server runs Jenkis CI, but this does not affect the installation, and you can use it with peace of mind, as only successful builds are released
The offline installers you mentioned are outdated, they are at v21.12 while the current versions are 22.04
Apparently your problem was when downloading the installer, and something went wrong
the problem was here
ERROR: Chocolatey expected a file at 'C:\Users\yen\AppData\Local\Temp\chocolatey\kdeconnect-kde\22.04.1.973\kdeconnect-kde-22.04.1-973-windows-msvc2019_64-cl.exe' to be of length '66510288' but the length was '30843341'.

In short: a 66MB file was expected but the downloaded one was only 30MB
And Chocolatey uses checksums to verify file integrity

Maybe there is a problem with the installation script

**Comment by yan12125** [comment]
> Yes, indeed their server runs Jenkis CI, but this does not affect the installation, and you can use it with peace of mind, as only successful builds are released

I assume a successful build on CI only means there are no compile errors. KDE developers do not guarantee its stability.

Also, from https://docs.chocolatey.org/en-us/create/create-packages#what-version-of-the-software-should-i-package,

```
The main release of a product versions are usually sufficient. If there are also beta versions available and you would rather have that, then please create both the official release and the beta (and set the beta as a prerelease when pushing the item to community.chocolatey.org). Regular users of packages may want to use official releases only and not betas.
```

Although CI-generated installers are more like "nightlies" than betas, I assume the suggestion still applies.

> The offline installers you mentioned are outdated, they are at v21.12 while the current versions are 22.04

Yeah, those logs are from the last version when I created this GitHub issue. I still got similar issues from newer versions from time to time.

> Maybe there is a problem with the installation script

When I got this issue in Chocolatey, other means of downloading (ex: via Firefox) fail as well. Maybe the stability of the CI server is a lower priority than the official download server https://download.kde.org/. As a result, I suggest to package stable versions, so that users don't need to bother with the CI server.

**Comment by opusforlife2** [comment]
I would also prefer going for a version that the KDE Connect team has specifically marked stable enough for a Windows release. For example, this is the latest stable version for Windows at the time of writing this comment: https://download.kde.org/stable/release-service/22.08.1/windows/

Edit: @AndreAugustoAAQ Ping, just in case you're not watching this repo.

**Comment by yan12125** [comment]
Upgrading kdeconnect-kde now comes with a different error:

> ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://binary-factory.kde.org/job/kdeconnect-kde_Release_win64/1156/artifact/kdeconnect-kde-22.08.3-1156-windows-msvc2019_64-cl.exe'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."

Update: the 404 issue is also reported in https://github.com/AndreAugustoAAQ/chocolatey-packages/issues/3

**Comment by AndreAugustoDev** [comment]
If someone proposes to develop the scripts for stable marked versions I can be reviewing and adding to the repository 😃

**Comment by opusforlife2** [comment]
Did some research on the other KDE Chocolatey packages. The URLs are almost always download links from the official websites of those specific apps. One package even points to the binary factory link, so you're not alone in going that route. :D

But one I found to be somewhat promising for our needs: [digiKam](https://github.com/tunisiano187/Chocolatey-packages/blob/master/automatic/digikam/update.ps1#L22)

However, I followed the [link](https://github.com/tunisiano187/Chocolatey-packages/blob/3850e4e7f5a69cc44dd2d2d42a598ee84d985420/automatic/digikam/update.ps1#L4) back to the KDE Connect [folder](https://download.kde.org/stable/kdeconnect/?C=M;O=D) and it was last updated 2 years ago.

So, finally, I've opened a new bug in the KDE Bugzilla. I asked them to keep the https://kdeconnect.kde.org/download.html page up to date with the latest build. That neatly solves our problem completely. No need to fuss around with URL regexes.

I also proposed an alternative, which is to keep the folder I linked above up to date.

Let's see what response I get.

**Comment by yan12125** [comment]
> So, finally, I've opened a new bug in the KDE Bugzilla.

Thanks! Is this the bug you created? https://bugs.kde.org/show_bug.cgi?id=463117

**Comment by opusforlife2** [comment]
Yep.

**Comment by opusforlife2** [comment]
@yan12125 Since you have access to the KDE Invent repo, could you ask the devs for guidance on how to find the URLs to the latest stable versions of the app? Maybe they provide tips. Maybe they decide to fix this themselves, who knows.

**Comment by yan12125** [comment]
I'm not sure if the repo https://invent.kde.org/websites/kdeconnect-kde-org is the best place to ask or not. I asked on #kde IRC channel instead. Someone mentions the latest version can be found on Microsoft Store. I cannot find a reliable way to get the URL out of Microsoft Store, though.

By the way, I remeber everyone can register an account for https://invent.kde.org/.

**Comment by opusforlife2** [comment]
Could you find out who's in charge of the KDE Connect MS Store uploads? They could point us in the right direction.

**Comment by yan12125** [comment]
No idea. https://apps.microsoft.com/store/detail/kde-connect/9N93MRMSXBF0 says the publisher is "KDE e.V".

**Comment by opusforlife2** [comment]
Well, it's obviously gotta be under the org's banner, but there must be _someone_ in the KDE Connect dev team who decides "hey this release looks stable enough, let's put it on MS Store."

**Comment by yan12125** [comment]
Probably. I don't have time for further investigation.

**Comment by opusforlife2** [comment]
@yan12125 Had a chat with the devs. The dev handling Windows left the team, so they need testers for Windows release candidates. They plan on one release every 4 months.

Only if the process gets streamlined will it be worth it to look at pointing this choco package at the official download page for stable releases.

**Comment by yan12125** [comment]
Thank you very much for following up. Any details about how to help testing?

**Comment by opusforlife2** [comment]
Since you have a KDE account, you could be pinged there. Or, you could join the KDE Connect Matrix room.

**Comment by yan12125** [comment]
When looking around for information about testing kdeconnect on Windows, I found https://invent.kde.org/network/kdeconnect-kde/-/blob/master/data/org.kde.kdeconnect.metainfo.xml (mentioned on https://community.kde.org/KDEConnect#Windows_Store). That seems a reliable source for information about the latest KDE Connect for Windows. By the way, here comes a new version 22.12.3.

**Comment by opusforlife2** [comment]
Come join the Matrix room. The devs are pretty active in it.

**Comment by yan12125** [comment]
Thanks. I guess I don't have any more questions - joined, anyway.

**Comment by yan12125** [comment]
Proposed the change at #4
